src/Components/CreateImageWizardV2/validators.ts

```diff
@@ -9,11 +9,10 @@ export const isAwsAccountIdValid = (awsAccountId: string | undefined) => {
   return false;
 };
 
-// TODO: this validator thinks asdf@asdf is a valid e-mail address, is that intentional or a bug?
 export const isGcpEmailValid = (gcpShareWithAccount: string | undefined) => {
   if (
     gcpShareWithAccount !== undefined &&
-    /^[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,}$/.test(gcpShareWithAccount)
+    /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,253}$/.test(gcpShareWithAccount)
```

Comment:
```suggestion
    /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,12}$/.test(gcpShareWithAccount) &&
    gcpShareWithAccount.length <= 253
```

This boundary applies only the to the last character set `[a-z]` so for example to the **.com** in `test@test.com`. I'd say we can make the upper limit way smaller. 

We could probably also add a max limit to the entire e-mail address, but that could be done with adding `.length` check to the validator which could be easier to read than adding the char limits into the regex.

Reply:
My 2 cents here, but there's actually an RFC that gives the proper regex to use to validate email addresses:
https://pdw.ex-parrot.com/Mail-RFC822-Address.html
there's a JS implementation apparently here https://gist.github.com/badsyntax/719800

It's worth having a look I think :slightly_smiling_face:

Reply:
https://stackoverflow.com/questions/20771794/mailrfc822address-regex has a good answer on the topic too.

Reply:
Ugh, I've just realized - is the validation case sensitive? Because I think it is and we don't really take this into account. 

```suggestion
    /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,12}$/.test(gcpShareWithAccount) &&
    gcpShareWithAccount.length <= 253
```

Reply:
I agree we should dig into this at some point, but perhaps not now... so much depends on shipping the V2 Wizard, I think we should avoid getting bogged down in details like this. I opened an issue and we can figure this all out in a follow-up PR. For now, would you all mind calling our minor improvement on what we already have in production a win and moving on?

https://github.com/RedHatInsights/image-builder-frontend/issues/1554

Reply:
Thanks for the issue, let's look at it later.